cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_resources.go

```diff
@@ -425,22 +424,6 @@ func (s *Server) inspectKappK8sResources(ctx context.Context, cluster, namespace
 	// As per https://github.com/vmware-tanzu/carvel-kapp-controller/blob/v0.32.0/pkg/deploy/kapp.go#L151
 	appName := fmt.Sprintf("%s-ctrl", packageId)
 
-	_, dynClient, err := s.GetClients(ctx, cluster)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "unable to get the k8s client: '%v'", err)
-	}
-
-	gvr := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "configmaps"}
-	resource := dynClient.Resource(gvr).Namespace(namespace)
-
-	// In order to be able to fetch the resources created by the kapp-controller, we need to fetch a ConfigMap
-	// that contains the label (the application id) used for query the k8s resources.
-	// We actively wait for this ConfigMap to be present in the cluster before returning the list of resources
```

Comment:
I wonder if should still wait for a couple of seconds (not infinitely), for the CM to be created in those cases everything is fine and it is just a matter of giving some time to k8s to actually create the resource.